javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserAnonymousClassDeclaration.java

```diff
@@ -142,15 +151,18 @@ public AccessSpecifier accessSpecifier() {
     return AccessSpecifier.PRIVATE;
   }
 
-  @Override
-  public List<ResolvedReferenceType> getAncestors(boolean acceptIncompleteList) {
-    return
-        ImmutableList.
-            <ResolvedReferenceType>builder()
-            .add(getSuperClass())
-            .addAll(superTypeDeclaration.asReferenceType().getAncestors(acceptIncompleteList))
-            .build();
-  }
+    @Override
+    public List<ResolvedReferenceType> getAncestors(boolean acceptIncompleteList) {
+        ImmutableList.Builder<ResolvedReferenceType> builder = ImmutableList.builder();
+
+        // Only add the super type if it is present (e.g. java.lang.Object has no super class)
+        getSuperClass().ifPresent(builder::add);
+
+        // All all ancestors of the super type..? TODO: Does this need to be wrapped in a presence check?
```

Comment:
Commenting for visibility...

My thinking now as I read through these changes is that this shouldn't need a presence check -- instead `getAncestors()` should return an empty collection if there are zero ancestors.

Reply:
Removed the todo... the above comment is correct that getAncestor returns an empty list, thus addall of an empty list will not cause an error